### PR TITLE
start restructuring work

### PR DESCRIPTION
**Pull request by maelle**

### CODE REVIEW

**Comment by maelle** [comment]
Goals = having a good code structure for the current functions, before adding others.

Tackles
#11 (a bit)
#1 
#30 
#4 (as much as I could at the moment)
#38

Regarding the removal of context in tests https://www.tidyverse.org/articles/2019/04/testthat-2-1-0/#context-is-now-optional

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1,3 @@
+* text=auto
+
+tests/fixtures/**/* -diff
```
Comment (maelle) [inline .gitattributes:3]: see https://github.com/ropensci/vcr/issues/109

File: .travis.yml [hunk 2]
```diff
@@ -2,6 +2,15 @@ language: r
 warnings_are_errors: true
 cache: packages
 
+addons:
+  apt:
+    sources:
+      - sourceline: 'ppa:chris-lea/libsodium'
+    packages:
+      - libsecret-1-dev
+      - libsodium-dev
```
Comment (maelle) [inline .travis.yml:11]: Needed for keyring

**Comment by maelle** [comment]
The failure on Mac is a LaTeX error :scream: :see_no_evil:

**Comment by stephlocke** [comment]
A NOTE in travis
>hubspot_key_get: no visible global function definition for ‘is’
Undefined global functions or variables:
  is
Consider adding
  importFrom("methods", "is")
to your NAMESPACE file (and ensure that your DESCRIPTION Imports field
contains 'methods').

Will review properly tomorrow probably. For now I've looked through all the files and things seem fine. I just want to trial the user experience :)

**Comment by maelle** [comment]
Hopefully the user experience didn't change! Do you think you'll have reviewed this before Monday? :-)

**Review by stephlocke — CHANGES_REQUESTED** [review]
Can you add the new set function to the example(s) in the README? Can still set to the value of demo

**Comment by maelle** [comment]
before you can review?

**Comment by stephlocke** [comment]
> before you can review?

nope, done my review and happy for the changes, just want to keep the docs in sync for when we merge in

**Comment by maelle** [comment]
Ok! Thanks!

**Pull request merged**